ExomeDepth/exomedepth_db.py

```diff
@@ -0,0 +1,231 @@
+        sample_update.refset = args.refset
+        session.add(sample_update)
+        session.commit()
+        print("## Changed refset of sample {0} with flowcell {1} to refset {2}".format(
+            args.sample_id, args.flowcell_id, args.refset)
+        )
+
+
+def print_all_samples(args):
+    Session = connect_database()
+    with Session() as session:
+        print("Name\tFlowcell\tRefset\tFamilyID")
+        for item in session.query(Sample).all():
+            print("{0}\t{1}\t{2}".format(item.sample, item.flowcell, item.refset))
+
+
+def query_refset(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        print(session.query(
+            Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+
+
+def get_flowcelid_bam(bam):
+    with pysam.AlignmentFile(bam, "rb") as workfile:
+        readgroups = []
+        for readgroup in workfile.header['RG']:
+            if readgroup['PU'] not in readgroup:
+                readgroups.append(readgroup['PU'])
+        readgroups = list(set(readgroups))
+        flowcell_id = "_".join(readgroups)
+    return flowcell_id
+
+
+def query_refset_bam(args):
+    Session = connect_database()
+    flowcell_id = get_flowcelid_bam(args.bam)
+    sample_id = determine_sample_id(args.bam)
+    with Session() as session:
+        print(session.query(
+            Sample).filter(Sample.sample == sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+
+
+def delete_sample_db(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        if session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).all():
+            session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).delete()
+            session.commit()
+            print("Deleted sample {0} with flowcell {1}.".format(args.sample_id, flowcell_id))
+        else:
+            print("Sample {0} with flowcell {1} not in database.".format(args.sample_id, flowcell_id))
+
+
+def get_flowcell_id(flowcellsarg):
+    flowcells = list(set(flowcellsarg))
+    flowcells.sort()
+    flowcell_id = "_".join(flowcells)
+    return flowcell_id
+
+
+def add_sample_to_db_and_return_refset(args):
+    Session = connect_database()
+    flowcell_id = get_flowcell_id(args.flowcell_id)
+    with Session() as session:
+        if not session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).all():
+            entry = create_sample(args.sample_id, flowcell_id, args.refset)
+            store_sample(Session, entry)
+        refset = (
+            session.query(Sample).filter(Sample.sample == args.sample_id).filter(Sample.flowcell == flowcell_id).one().refset
+        )
+        print(refset)
+        return refset
```

Comment:
No, if sample is already in database, it will set refset to the one in the database, and will not use argument.
If sample is not present in database, then it will use argument refset.

Will clarify the naming

Reply:
add_sample_to_db_and_return_refset is a bit of legacy function as add_sample_to_db_and_return_refset_bam is used in DxNextflowWES and igv_xml_session.py.

Removed add_sample_to_db_and_return_refset completely